src/main/java/org/opensearch/security/configuration/SettingsPermissionValve.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.configuration;
+
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.security.privileges.PrivilegesEvaluationContext;
+
+public interface SettingsPermissionValve {
```

Comment:
The presence of the `DlsFlsRequestValve` has mostly historical reasons, as it was once an optional module. 

I do not believe we need to make the settings access control optional (do we?). If we do not have the need to make it optional, I would recommend to make it non-pluggable to reduce complexity.